File: app/src/main/java/com/brainwellnessspa/MembershipModule/Activities/MembershipActivity.java
package com.brainwellnessspa.MembershipModule.Activities;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.brainwellnessspa.BWSApplication;
import com.brainwellnessspa.FaqModule.Models.FaqListModel;
import com.brainwellnessspa.MembershipModule.Adapters.MembershipPlanAdapter;
import com.brainwellnessspa.MembershipModule.Adapters.SubscriptionAdapter;
import com.brainwellnessspa.MembershipModule.Models.MembershipPlanListModel;
import com.brainwellnessspa.R;
import com.brainwellnessspa.Utility.APIClient;
import com.brainwellnessspa.Utility.CONSTANTS;
import com.brainwellnessspa.Utility.MeasureRatio;
import com.brainwellnessspa.WebView.TncActivity;
import com.brainwellnessspa.databinding.ActivityMembershipBinding;
import com.brainwellnessspa.databinding.AudioFaqLayoutBinding;
import com.segment.analytics.Properties;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MembershipActivity extends AppCompatActivity {
    ActivityMembershipBinding binding;
    SubscriptionAdapter subscriptionAdapter;
    MembershipPlanAdapter membershipPlanAdapter;
    Context ctx;
    MembershipFaqAdapter adapter;
    private long mLastClickTime = 0;
    Activity activity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_membership);
        ctx = MembershipActivity.this;
        activity = MembershipActivity.this;

        binding.llBack.setOnClickListener(view -> {
//            Intent i = new Intent(ctx, LoginActivity.class);
//            startActivity(i);
//            finish();
        });

        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(ctx, LinearLayoutManager.HORIZONTAL, false);
        binding.rvList.setLayoutManager(mLayoutManager);
        binding.rvList.setItemAnimator(new DefaultItemAnimator());

        RecyclerView.LayoutManager mLayoutManager1 = new LinearLayoutManager(ctx, LinearLayoutManager.VERTICAL, false);
        binding.rvPlanList.setLayoutManager(mLayoutManager1);
        binding.rvPlanList.setItemAnimator(new DefaultItemAnimator());

        RecyclerView.LayoutManager serachList = new LinearLayoutManager(ctx, LinearLayoutManager.VERTICAL, false);
        binding.rvFaqList.setLayoutManager(serachList);
        binding.rvFaqList.setItemAnimator(new DefaultItemAnimator());

        Properties p = new Properties();
        BWSApplication.addToSegment("Plan List Viewed", p, CONSTANTS.screen);

        if (BWSApplication.isNetworkConnected(this)) {
            BWSApplication.showProgressBar(binding.progressBar, binding.progressBarHolder, activity);
            Call<FaqListModel> listCall = APIClient.getClient().getFaqListings();
            listCall.enqueue(new Callback<FaqListModel>() {
                @Override
                public void onResponse(Call<FaqListModel> call, Response<FaqListModel> response) {
                    try {
                        if (response.isSuccessful()) {
                            BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, activity);
                            FaqListModel listModel = response.body();
                            binding.tvFaqTitle.setText(R.string.f_A_Q);
                            adapter = new MembershipFaqAdapter(listModel.getResponseData(), ctx, binding.rvFaqList, binding.tvFound);
                            binding.rvFaqList.setAdapter(adapter);
                        }
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }

                @Override
                public void onFailure(Call<FaqListModel> call, Throwable t) {
                    BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, activity);
                }
            });
        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), activity);
        }
    }

    @Override
    protected void onResume() {
        prepareMembershipData();
        super.onResume();
    }

    @Override
    public void onBackPressed() {
//        Intent i = new Intent(ctx, LoginActivity.class);
//        startActivity(i);
//        finish();
    }

    private void prepareMembershipData() {
        if (BWSApplication.isNetworkConnected(this)) {
            BWSApplication.showProgressBar(binding.progressBar, binding.progressBarHolder, activity);
            Call<MembershipPlanListModel> listCall = APIClient.getClient().getMembershipPlanList();
            listCall.enqueue(new Callback<MembershipPlanListModel>() {
                @Override
                public void onResponse(Call<MembershipPlanListModel> call, Response<MembershipPlanListModel> response) {
                    try {
                        if (response.isSuccessful()) {
                            BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, activity);
                            MembershipPlanListModel membershipPlanListModel = response.body();
                            if (membershipPlanListModel.getResponseCode().equalsIgnoreCase(getString(R.string.ResponseCodesuccess))) {
                                binding.btnFreeJoin.setVisibility(View.VISIBLE);
                                binding.tvTitle.setText(membershipPlanListModel.getResponseData().getTitle());
                                binding.tvDesc.setText(membershipPlanListModel.getResponseData().getDesc());
                                binding.tvTag.setText(R.string.membership_title);
                                binding.tvText.setText(getString(R.string.privacy_policy_t_n_c));
                                binding.tvtncs.setText(getString(R.string.t_n_csm));
                                binding.tvPrivacyPolicys.setText(getString(R.string.privacy_policysm));
                                binding.tvAnd.setText(getString(R.string.and));
                                binding.tvDisclaimers.setText(R.string.disclaimers);
                                binding.tvtncs.getPaint().setUnderlineText(true);
                                binding.tvPrivacyPolicys.getPaint().setUnderlineText(true);
                                binding.tvDisclaimers.getPaint().setUnderlineText(true);
                                binding.tvtncs.setOnClickListener(view -> {
                                    Intent i = new Intent(ctx, TncActivity.class);
                                    i.putExtra(CONSTANTS.Web, "Tnc");
                                    startActivity(i);
                                });

                                binding.tvPrivacyPolicys.setOnClickListener(view -> {
                                    Intent i = new Intent(ctx, TncActivity.class);
                                    i.putExtra(CONSTANTS.Web, "PrivacyPolicy");
                                    startActivity(i);
                                });

                                binding.tvDisclaimers.setOnClickListener(view -> {
                                    final Dialog dialog = new Dialog(ctx);
                                    dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
                                    dialog.setContentView(R.layout.full_desc_layout);
                                    dialog.getWindow().setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.dark_blue_gray)));
                                    dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
                                    final TextView tvTitle = dialog.findViewById(R.id.tvTitle);
                                    final TextView tvDesc = dialog.findViewById(R.id.tvDesc);
                                    final RelativeLayout tvClose = dialog.findViewById(R.id.tvClose);
                                    tvTitle.setText(R.string.Disclaimer);
                                    tvDesc.setText(R.string.Disclaimer_text);
                                    dialog.setOnKeyListener((v, keyCode, event) -> {
                                        if (keyCode == KeyEvent.KEYCODE_BACK) {
                                            dialog.dismiss();
                                            return true;
                                        }
                                        return false;
                                    });

                                    tvClose.setOnClickListener(v -> dialog.dismiss());
                                    dialog.show();
                                    dialog.setCancelable(false);
                                });
                                MeasureRatio measureRatio = BWSApplication.measureRatio(ctx, 0,
                                        5, 3, 1f, 0);
                                binding.ivRestaurantImage.getLayoutParams().height = (int) (measureRatio.getHeight() * measureRatio.getRatio());
                                binding.ivRestaurantImage.getLayoutParams().width = (int) (measureRatio.getWidthImg() * measureRatio.getRatio());
                                binding.ivRestaurantImage.setScaleType(ImageView.ScaleType.FIT_XY);
                                binding.ivRestaurantImage.setImageResource(R.drawable.ic_membership_banner);

                                membershipPlanAdapter = new MembershipPlanAdapter(membershipPlanListModel.getResponseData().getPlan(), ctx, binding.btnFreeJoin,
                                        membershipPlanListModel.getResponseData().getTrialPeriod(), activity);
                                binding.rvPlanList.setAdapter(membershipPlanAdapter);

//                                subscriptionAdapter = new SubscriptionAdapter(membershipPlanListModel.getResponseData().getAudioFiles(), ctx);
//                                binding.rvList.setAdapter(subscriptionAdapter);
                            }
                        }
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }

                @Override
                public void onFailure(Call<MembershipPlanListModel> call, Throwable t) {
                    BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, activity);
                }
            });
        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), activity);
        }
    }

    public class MembershipFaqAdapter extends RecyclerView.Adapter<MembershipFaqAdapter.MyViewHolder> {
        private List<FaqListModel.ResponseData> modelList;
        Context ctx;
        RecyclerView rvFaqList;
        TextView tvFound;

        public MembershipFaqAdapter(List<FaqListModel.ResponseData> modelList, Context ctx, RecyclerView rvFaqList, TextView tvFound) {
            this.modelList = modelList;
            this.ctx = ctx;
            this.rvFaqList = rvFaqList;
            this.tvFound = tvFound;
        }

        @NonNull
        @Override
        public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            AudioFaqLayoutBinding v = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext())
                    , R.layout.audio_faq_layout, parent, false);
            return new MyViewHolder(v);
        }

        @Override
        public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
            holder.binding.tvTitle.setText(modelList.get(position).getTitle());
            holder.binding.tvDesc.setText(modelList.get(position).getDesc());

            holder.binding.ivClickRight.setOnClickListener(view -> {
                holder.binding.llMainLayout.setBackgroundResource(R.color.discalimer_gray);
                holder.binding.tvDesc.setFocusable(true);
                holder.binding.tvDesc.requestFocus();
                holder.binding.tvDesc.setVisibility(View.VISIBLE);
                holder.binding.ivClickRight.setVisibility(View.GONE);
                holder.binding.ivClickDown.setVisibility(View.VISIBLE);
                holder.binding.ivClickDown.setImageResource(R.drawable.ic_down_black_icon);
            });

            holder.binding.ivClickDown.setOnClickListener(view -> {
                holder.binding.llMainLayout.setBackgroundResource(R.color.white);
                holder.binding.tvDesc.setVisibility(View.GONE);
                holder.binding.ivClickRight.setVisibility(View.VISIBLE);
                holder.binding.ivClickDown.setVisibility(View.GONE);
                holder.binding.ivClickDown.setImageResource(R.drawable.ic_back_black_icon);
            });

            if (modelList.size() == 0) {
                tvFound.setVisibility(View.GONE);
                rvFaqList.setVisibility(View.GONE);
            } else {
                tvFound.setVisibility(View.GONE);
                rvFaqList.setVisibility(View.VISIBLE);
            }
        }

        @Override
        public int getItemCount() {
            return modelList.size();
        }

        public class MyViewHolder extends RecyclerView.ViewHolder {
            AudioFaqLayoutBinding binding;

            public MyViewHolder(AudioFaqLayoutBinding binding) {
                super(binding.getRoot());
                this.binding = binding;
            }
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/membershipModule/activities/MembershipActivity.kt
package com.brainwellnessspa.membershipModule.activities

import android.annotation.SuppressLint
import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.*
import android.widget.Button
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.brainwellnessspa.BWSApplication
import com.brainwellnessspa.BWSApplication.callSignActivity
import com.brainwellnessspa.R
import com.brainwellnessspa.dashboardModule.models.PlanlistInappModel
import com.brainwellnessspa.databinding.ActivityMembershipBinding
import com.brainwellnessspa.databinding.AudioFaqLayoutBinding
import com.brainwellnessspa.databinding.MembershipPlanBinding
import com.brainwellnessspa.databinding.SubscribeBoxLayoutBinding
import com.brainwellnessspa.faqModule.models.FaqListModel
import com.brainwellnessspa.membershipModule.models.MembershipPlanListModel
import com.brainwellnessspa.utility.APINewClient.client
import com.brainwellnessspa.utility.CONSTANTS
import com.brainwellnessspa.utility.MeasureRatio
import com.brainwellnessspa.webView.TncActivity
import com.bumptech.glide.Glide
import com.bumptech.glide.Priority
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.google.gson.Gson
import com.segment.analytics.Properties
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class MembershipActivity : AppCompatActivity() {
    lateinit var binding: ActivityMembershipBinding
    var subscriptionAdapter: SubscriptionAdapter? = null
    var membershipPlanAdapter: MembershipPlanAdapter? = null
    lateinit var ctx: Context
    var adapter: MembershipFaqAdapter? = null
    private val mLastClickTime: Long = 0
    lateinit var i: Intent
    private var doubleBackToExitPressedOnce = false
    lateinit var act: Activity
    var UserId :String = ""
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_membership)
        ctx = this@MembershipActivity
        act = this@MembershipActivity

        binding.llBack.setOnClickListener { view ->
            callSignActivity(act)
        }
        val shared = getSharedPreferences(CONSTANTS.PREFE_ACCESS_SIGNIN_COUSER, Context.MODE_PRIVATE)
        UserId = shared.getString(CONSTANTS.PREFE_ACCESS_UserId, "")!!
        val mLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(ctx, LinearLayoutManager.HORIZONTAL, false)
        binding.rvList.layoutManager = mLayoutManager
        binding.rvList.itemAnimator = DefaultItemAnimator()
        val mLayoutManager1: RecyclerView.LayoutManager = LinearLayoutManager(ctx, LinearLayoutManager.VERTICAL, false)
        binding.rvPlanList.layoutManager = mLayoutManager1
        binding.rvPlanList.itemAnimator = DefaultItemAnimator()
        val serachList: RecyclerView.LayoutManager = LinearLayoutManager(ctx, LinearLayoutManager.VERTICAL, false)
        binding.rvFaqList.layoutManager = serachList
        binding.rvFaqList.itemAnimator = DefaultItemAnimator()
        val p = Properties()
        i = Intent(ctx, OrderSummaryActivity::class.java)
        i.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_NO_ANIMATION
        BWSApplication.addToSegment("Plan List Viewed", p, CONSTANTS.screen)
        if (BWSApplication.isNetworkConnected(this)) {
            BWSApplication.showProgressBar(binding.progressBar, binding.progressBarHolder, act)
            val listCall: Call<FaqListModel> = client.faqLists
            listCall.enqueue(object : Callback<FaqListModel?> {
                override fun onResponse(call: Call<FaqListModel?>?, response: Response<FaqListModel?>) {
                    try {
                        if (response.isSuccessful) {
                            BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, act)
                            val listModel: FaqListModel? = response.body()
                            binding.tvFaqTitle.text = getString(R.string.f_A_Q)
                            adapter = MembershipFaqAdapter(listModel!!.responseData!!, ctx, binding.rvFaqList, binding.tvFound)
                            binding.rvFaqList.adapter = adapter
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }

                override fun onFailure(call: Call<FaqListModel?>?, t: Throwable?) {
                    BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, act)
                }
            })
        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), this)
        }
    }

    override fun onResume() {
        prepareMembershipData()
        super.onResume()
    }

    override fun onBackPressed() {
        if (doubleBackToExitPressedOnce) {
            finishAffinity()
            return
        }
        this.doubleBackToExitPressedOnce = true
        BWSApplication.showToast("Press again to exit", act)

        Handler(Looper.myLooper()!!).postDelayed({
            doubleBackToExitPressedOnce = false
        }, 2000)
    }

    private fun prepareMembershipData() {
        if (BWSApplication.isNetworkConnected(this)) {
            BWSApplication.showProgressBar(binding.progressBar, binding.progressBarHolder, act)
            val listCall: Call<MembershipPlanListModel> = client.getMembershipPlanList()
            listCall.enqueue(object : Callback<MembershipPlanListModel?> {
                override fun onResponse(call: Call<MembershipPlanListModel?>?, response: Response<MembershipPlanListModel?>) {
                    try {
                        if (response.isSuccessful) {
                            BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, act)
                            val membershipPlanListModel: MembershipPlanListModel = response.body()!!
                            if (membershipPlanListModel.responseCode.equals(getString(R.string.ResponseCodesuccess))) {
                                binding.btnFreeJoin.visibility = View.VISIBLE
                                binding.tvTitle.text = membershipPlanListModel.responseData!!.title
                                binding.tvDesc.text = membershipPlanListModel.responseData!!.desc
                                binding.tvTag.setText(R.string.membership_title)
                                binding.tvText.text = getString(R.string.privacy_policy_t_n_c)
                                binding.tvtncs.text = getString(R.string.t_n_csm)
                                binding.tvPrivacyPolicys.text = getString(R.string.privacy_policysm)
                                binding.tvAnd.text = getString(R.string.and)
                                binding.tvDisclaimers.setText(R.string.disclaimers)
                                binding.tvtncs.paint.isUnderlineText = true
                                binding.tvPrivacyPolicys.paint.isUnderlineText = true
                                binding.tvDisclaimers.paint.isUnderlineText = true
                                binding.tvtncs.setOnClickListener {
                                    val i = Intent(ctx, TncActivity::class.java)
                                    i.putExtra(CONSTANTS.Web, "Tnc")
                                    startActivity(i)
                                }
                                binding.tvPrivacyPolicys.setOnClickListener {
                                    val i = Intent(ctx, TncActivity::class.java)
                                    i.putExtra(CONSTANTS.Web, "PrivacyPolicy")
                                    startActivity(i)
                                }
                                binding.tvDisclaimers.setOnClickListener {
                                    val dialog = Dialog(ctx)
                                    dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
                                    dialog.setContentView(R.layout.full_desc_layout)
                                    dialog.window!!.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.dark_blue_gray)))
                                    dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
                                    val tvTitle = dialog.findViewById<TextView>(R.id.tvTitle)
                                    val tvDesc = dialog.findViewById<TextView>(R.id.tvDesc)
                                    val tvClose = dialog.findViewById<RelativeLayout>(R.id.tvClose)
                                    tvTitle.setText(R.string.Disclaimer)
                                    tvDesc.setText(R.string.Disclaimer_text)
                                    dialog.setOnKeyListener { _: DialogInterface?, keyCode: Int, _: KeyEvent? ->
                                        if (keyCode == KeyEvent.KEYCODE_BACK) {
                                            dialog.dismiss()
                                            return@setOnKeyListener true
                                        }
                                        false
                                    }
                                    tvClose.setOnClickListener { v: View? -> dialog.dismiss() }
                                    dialog.show()
                                    dialog.setCancelable(false)
                                }
                                val measureRatio: MeasureRatio = BWSApplication.measureRatio(ctx, 0f, 5f, 3f, 1f, 0f)
                                binding.ivRestaurantImage.layoutParams.height = (measureRatio.height * measureRatio.ratio).toInt()
                                binding.ivRestaurantImage.layoutParams.width = (measureRatio.widthImg * measureRatio.ratio).toInt()
                                binding.ivRestaurantImage.scaleType = ImageView.ScaleType.FIT_XY
                                binding.ivRestaurantImage.setImageResource(R.drawable.ic_membership_banner)
                                membershipPlanAdapter = MembershipPlanAdapter(membershipPlanListModel.responseData!!.plan!!, ctx, binding.btnFreeJoin, membershipPlanListModel.responseData!!.trialPeriod!!, act, i)
                                binding.rvPlanList.adapter = membershipPlanAdapter
                                subscriptionAdapter = SubscriptionAdapter(membershipPlanListModel.responseData!!.audioFiles!!, ctx)
                                binding.rvList.adapter = subscriptionAdapter
                            }
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }

                override fun onFailure(call: Call<MembershipPlanListModel?>?, t: Throwable?) {
                    BWSApplication.hideProgressBar(binding.progressBar, binding.progressBarHolder, act)
                }
            })
        } else {
            BWSApplication.showToast(getString(R.string.no_server_found), this)
        }
    }

    inner class MembershipFaqAdapter(var modelList: List<FaqListModel.ResponseData>, var ctx: Context, var rvFaqList: RecyclerView, var tvFound: TextView) : RecyclerView.Adapter<MembershipFaqAdapter.MyViewHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            val v: AudioFaqLayoutBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.audio_faq_layout, parent, false)
            return MyViewHolder(v)
        }

        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            holder.binding.tvTitle.text = modelList[position].title
            holder.binding.tvDesc.text = modelList[position].desc
            holder.binding.ivClickRight.setOnClickListener {
                holder.binding.llMainLayout.setBackgroundResource(R.color.discalimer_gray)
                holder.binding.tvDesc.isFocusable = true
                holder.binding.tvDesc.requestFocus()
                holder.binding.tvDesc.visibility = View.VISIBLE
                holder.binding.ivClickRight.visibility = View.GONE
                holder.binding.ivClickDown.visibility = View.VISIBLE
                holder.binding.ivClickDown.setImageResource(R.drawable.ic_down_arrow_icon)
            }
            holder.binding.ivClickDown.setOnClickListener { view ->
                holder.binding.llMainLayout.setBackgroundResource(R.color.white)
                holder.binding.tvDesc.visibility = View.GONE
                holder.binding.ivClickRight.visibility = View.VISIBLE
                holder.binding.ivClickDown.visibility = View.GONE
                holder.binding.ivClickDown.setImageResource(R.drawable.ic_back_black_icon)
            }
            if (modelList.isEmpty()) {
                tvFound.visibility = View.GONE
                rvFaqList.visibility = View.GONE
            } else {
                tvFound.visibility = View.GONE
                rvFaqList.visibility = View.VISIBLE
            }
        }

        override fun getItemCount(): Int {
            return modelList.size
        }

        inner class MyViewHolder(var binding: AudioFaqLayoutBinding) : RecyclerView.ViewHolder(binding.root)
    }


    class SubscriptionAdapter(private val listModelList: ArrayList<MembershipPlanListModel.AudioFile>, var ctx: Context) : RecyclerView.Adapter<SubscriptionAdapter.MyViewHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            val v: SubscribeBoxLayoutBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.subscribe_box_layout, parent, false)
            return MyViewHolder(v)
        }

        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            val listModel = listModelList[position]
            holder.binding.tvTitle.text = listModel.name
            Glide.with(ctx).load(listModel.imageFile).thumbnail(0.05f).apply(RequestOptions.bitmapTransform(RoundedCorners(12))).priority(Priority.HIGH).diskCacheStrategy(DiskCacheStrategy.ALL).skipMemoryCache(false).into(holder.binding.ivRestaurantImage)
        }

        override fun getItemCount(): Int {
            return listModelList.size
        }

        inner class MyViewHolder(var binding: SubscribeBoxLayoutBinding) : RecyclerView.ViewHolder(binding.root)
    }

    //import com.brainwellnessspa.MembershipModule.Activities.OrderSummaryActivity;
    class MembershipPlanAdapter(private val listModelList: ArrayList<MembershipPlanListModel.Plan>, var ctx: Context, var btnFreeJoin: Button, var TrialPeriod: String, var activity: Activity, var i: Intent) : RecyclerView.Adapter<MembershipPlanAdapter.MyViewHolder>() {
        private var rowIndex = -1
        private var pos = 0
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            val v: MembershipPlanBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.membership_plan, parent, false)
            return MyViewHolder(v)
        }

        @SuppressLint("SetTextI18n") override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            val listModel = listModelList[position] //        holder.binding.tvTitle.setText(listModel.getTitle());
            holder.binding.tvPlanFeatures01.text = listModel.planFeatures?.get(0)?.feature
            holder.binding.tvPlanFeatures02.text = listModel.planFeatures?.get(1)?.feature
            holder.binding.tvPlanFeatures03.text = listModel.planFeatures?.get(2)?.feature
            holder.binding.tvPlanFeatures04.text = listModel.planFeatures?.get(3)?.feature
            holder.binding.tvPlanAmount.text = "$" + listModel.planAmount
            holder.binding.tvSubName.text = listModel.subName
            holder.binding.tvPlanInterval.text = listModel.planInterval
            if (listModel.recommendedFlag.equals("1", ignoreCase = true)) {
                holder.binding.tvRecommended.visibility = View.VISIBLE/*  if (pos == 0) {
                holder.binding.llPlanSub.setBackgroundColor(ctx.getResources().getColor(R.color.blue));
                holder.binding.llFeatures.setVisibility(View.VISIBLE);
                holder.binding.tvPlanAmount.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvSubName.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.tvPlanInterval.setTextColor(ctx.getResources().getColor(R.color.white));
                holder.binding.llFeatures.setBackgroundColor(ctx.getResources().getColor(R.color.white));
            }*/
            } else {
                holder.binding.tvRecommended.visibility = View.GONE
            }
            holder.binding.llPlanMain.setOnClickListener {
                rowIndex = position
                pos++
                notifyDataSetChanged()
            }
            if (rowIndex == position) {
                changeFunction(holder, listModel, position)
            } else {
                if (listModel.recommendedFlag.equals("1", ignoreCase = true) && pos == 0) {
                    holder.binding.tvRecommended.visibility = View.VISIBLE
                    changeFunction(holder, listModel, position)
                } else {
                    holder.binding.llPlanSub.background = ContextCompat.getDrawable(activity, R.drawable.rounded_light_gray)
                    holder.binding.tvPlanAmount.setTextColor(ContextCompat.getColor(activity, R.color.black))
                    holder.binding.tvSubName.setTextColor(ContextCompat.getColor(activity, R.color.black))
                    holder.binding.tvPlanInterval.setTextColor(ContextCompat.getColor(activity, R.color.black))
                    holder.binding.llFeatures.visibility = View.GONE
                }
            }
            btnFreeJoin.setOnClickListener {
                if (BWSApplication.isNetworkConnected(ctx)) {
                    activity.startActivity(i)
                    activity.finish()
                } else {
                    BWSApplication.showToast(ctx.getString(R.string.no_server_found), activity)
                }
            }
        }

        private fun changeFunction(holder: MyViewHolder, listModel: MembershipPlanListModel.Plan, position: Int) {
            holder.binding.llPlanSub.setBackgroundResource(R.drawable.top_round_green_cornor)
            holder.binding.llFeatures.visibility = View.VISIBLE
            holder.binding.tvPlanAmount.setTextColor(ContextCompat.getColor(activity, R.color.white))
            holder.binding.tvSubName.setTextColor(ContextCompat.getColor(activity, R.color.white))
            holder.binding.tvPlanInterval.setTextColor(ContextCompat.getColor(activity, R.color.white))
            holder.binding.llFeatures.setBackgroundColor(ContextCompat.getColor(activity, R.color.white))
            planFlag = listModel.planFlag.toString()
            price = listModel.planAmount.toString()
            planId = listModel.planID.toString()
            val gson = Gson();
            i = Intent(ctx, OrderSummaryActivity::class.java)
            i.putExtra("PlanData", gson.toJson(listModelList))
            i.putExtra("TrialPeriod", "")
            i.putExtra("position", position)
            i.putExtra("Promocode", "")
        }

        override fun getItemCount(): Int {
            return listModelList.size
        }

        inner class MyViewHolder(var binding: MembershipPlanBinding) : RecyclerView.ViewHolder(binding.root)

        companion object {
            @JvmField var planFlag = ""

            @JvmField var planId = ""

            @JvmField var price = ""
        }
    }

}